Modules/Sources/WordPressUI/Extensions/UIView+AutoLayout.swift

```diff
@@ -0,0 +1,100 @@
+import UIKit
+import SwiftUI
+
+extension UIView {
+    /// Pins edges of the view to the edges of the given container. By default,
+    /// pins to the superview.
+    @discardableResult
+    public func pinEdges(
+        _ edges: Edge.Set = .all,
+        to container: AutoLayoutItem? = nil,
+        insets: UIEdgeInsets = .zero,
+        relation: AutoLayoutPinEdgesRelation = .equal,
+        priority: UILayoutPriority? = nil
+    ) -> [NSLayoutConstraint] {
+        guard let container = container ?? superview else {
+            assertionFailure("view has to be installed in the view hierarchy")
+            return []
+        }
+        translatesAutoresizingMaskIntoConstraints = false
```

Comment:
> make sure self is a subview of container

It's not a strict requirement. You can pin a view to a layout guide or to a sibling view. It should be pretty clear from the method's name what it does. SnapKit, PureLayout, and a couple of other frameworks all pretty much work the same way, so it shouldn't surprise that that's how it works.

Reply:
If there is no hierarchical requirement in these two views, we should remove this line, as I mentioned in the example code above:

```swift
// Not okay to set `superview.translatesAutoresizingMaskIntoConstraints = false`
// because we don't know if superview's frame is set by auto-layout or not.
superview.pinEdges(to: subview)
```

Reply:
IIRC UIViewController.view is not framed using auto layout. Maybe you can reproduce the issue I mentioned by using `pinEdges(to: subview)` on a `UIViewController.view`.

Reply:
> As a user, you want to avoid writing translatesAutoresizingMaskIntoConstraints. [...] If you take pretty much every Auto Layout framework, they are designed to ensure you never have to write translatesAutoresizingMaskIntoConstraints.

That's true. I get it's a bit annoying that we are forced to write this one long line for pretty much every single view in the app. However, the framework (or convenient method in this case) should only set the property when it's appropriate to do so. As I mentioned above, it's not appropriate to call `translatesAutoresizingMaskIntoConstraints = true`, if `self` is not framed using auto-layout.

Here is a diff to reproduce the issue:

```diff
diff --git a/WordPress/Classes/ViewRelated/Reader/ReaderDiscoverViewController.swift b/WordPress/Classes/ViewRelated/Reader/ReaderDiscoverViewController.swift
index af9eda30d5..ffa871d863 100644
--- a/WordPress/Classes/ViewRelated/Reader/ReaderDiscoverViewController.swift
+++ b/WordPress/Classes/ViewRelated/Reader/ReaderDiscoverViewController.swift
@@ -101,6 +101,11 @@ class ReaderDiscoverViewController: UIViewController, ReaderDiscoverHeaderViewDe
         view.addSubview(streamVC.view)
         streamVC.view.pinEdges()
         streamVC.didMove(toParent: self)
+
+        let testView = UIView()
+        testView.backgroundColor = .red
+        view.insertSubview(testView, at: 0)
+        view.pinEdges(to: testView) // Reader goes blank.
     }
 
     /// TODO: (tech-debt) the app currently stores the responses from the `/discover`
```

Reply:
> Here is a diff to reproduce the issue:

I added a runtime check to catch this early:

```swift
#if DEBUG
        if let target = target as? UIView {
            assert(!target.isDescendant(of: self), "The target view can't be a descendant for the view")
        }
#endif
```